GetIntoTeachingApi/Controllers/TeachingEventsController.cs

```diff
@@ -163,6 +163,37 @@ public async Task<IActionResult> Get([FromRoute, SwaggerParameter("The `readable
             return Ok(new TeachingEventAddAttendee(candidate));
         }
 
+        [HttpPost]
+        [Route("")]
+        [SwaggerOperation(
+            Summary = "Adds or updates a teaching event.",
+            Description = "If the `id` is specified then the existing teaching event will be " +
+                          "updated, otherwise a new teaching event will be created.",
+            OperationId = "AddOrUpdateTeachingEvent",
+            Tags = new[] { "Teaching Events" })]
+        [ProducesResponseType(typeof(TeachingEvent), StatusCodes.Status201Created)]
+        [ProducesResponseType(typeof(IDictionary<string, string>), StatusCodes.Status400BadRequest)]
+        public async Task<IActionResult> AddOrUpdateTeachingEventAsync([FromBody] TeachingEvent teachingEvent)

```

Comment:
Can we call thIs method `Upsert()`? I use that terminology in the candidate job and should probably start changing the others to match, so `UpsertMember` and `UpsertAttendee` etc. Then the operation id can be `UpsertTeachingEvent`.